### PR TITLE
A0-2776: Bump rust version to 1.68.2

### PR DESCRIPTION
**Pull request by ggawryal**
# Description

Bumped rust to stable 1.68.2. 
Aleph-client and crates depending on it still have to use nightly, because it uses nightly `auto trait` feature.

Unfortunately, `cargo clippy` introduced a new warning `variables can be used directly in the format string`, which required modifying all `format!`s, `panic!`s and `print!`s (logging formatting is not linted for some reason). All changes in *.rs files were autogenerated using `cargo clippy --fix`, so feel free to just skim through it.

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- I have made necessary updates to the Infrastructure
- I have bumped aleph-client version if relevant

### CODE REVIEW

**Review by kostekIV — COMMENTED** [review]
Looks good, so now we have nightly for aleph-client and its consumers plus for cargo fmt? We could keep the version used in gh workflows and local scripts in one place, maybe in some `env` file in .github/scripts/.

File: .github/workflows/_check-code-formatting.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  workflow_call:
+
+jobs:
+  main:
+    name: Run check, test and lints
+    runs-on: [self-hosted, Linux, X64, large]
+    env:
+      CARGO_INCREMENTAL: 0
+      RUSTC_WRAPPER: sccache
+      RUST_TOOLCHAIN: nightly-2023-01-10
```
Comment (Marcin-Radecki) [inline .github/workflows/_check-code-formatting.yml:14]: I've created repository variable `RUST_TOOLCHAIN_NIGHTLY` with value `nightly-2023-01-10`. We can use it in CI context, e.g. in this file as `${{ vars.RUST_TOOLCHAIN_NIGHTLY }}`

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "nightly-2022-10-30"
```
Comment (Marcin-Radecki) [inline rust-toolchain.toml:2]: Please update root `Cargo.lock` file

**Review by Marcin-Radecki — APPROVED** [review]
Nice job!

File: aleph-client/rust-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
+channel = "nightly-2023-01-10"
+targets = [ "wasm32-unknown-unknown" ]
+components = [ "rustfmt", "clippy", "rust-src" ]
+profile = "minimal"
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply aleph-client/rust-toolchain.toml]: done x5 (+ bonus one in contracts/rust-toolchain.toml)

File: scripts/run_checks_on_aleph_node.sh [hunk 4]
```diff
@@ -3,5 +3,5 @@
 set -e
 
 CARGO_INCREMENTAL=0 cargo clippy --all-targets --all-features -- --no-deps -D warnings
-CARGO_INCREMENTAL=0 cargo fmt --all
-CARGO_INCREMENTAL=0 cargo test --lib
+CARGO_INCREMENTAL=0 cargo +nightly-2023-01-10 fmt --all
+CARGO_INCREMENTAL=0 cargo test --lib
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply scripts/run_checks_on_aleph_node.sh]: added too

File: rust-toolchain.toml [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
+channel = "stable-2023-03-28"
```
[the remark this answers is not in the document]
Comment (ggawryal) [reply rust-toolchain.toml]: Good idea

File: rust-toolchain.toml (hunk 2 above)
Comment (ggawryal) [reply rust-toolchain.toml:2]: Great catch, updated

**Review by kostekIV — APPROVED** [review]

**Pull request merged**